data/emulator.js

```diff
@@ -1880,6 +1885,41 @@ class EmulatorJS {
             }
         });
         
+        this.diskParent = this.createElement("div");
+        this.diskParent.id = "ejs_disksMenu";
+        this.disksMenuOpen = false;
+        const diskButton = addButton("Disks", '<svg fill="#FFFFFF" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 473.109 473.109"><path d="M340.963,101.878H12.105C5.423,101.878,0,107.301,0,113.983v328.862c0,6.68,5.423,12.105,12.105,12.105h328.857 c6.685,0,12.104-5.426,12.104-12.105V113.983C353.067,107.301,347.647,101.878,340.963,101.878z M67.584,120.042h217.895v101.884 H67.584V120.042z M296.076,429.228H56.998V278.414h239.079V429.228z M223.947,135.173h30.269v72.638h-30.269V135.173z M274.13,315.741H78.933v-12.105H274.13V315.741z M274.13,358.109H78.933v-12.105H274.13V358.109z M274.13,398.965H78.933v-12.105 H274.13V398.965z M473.109,30.263v328.863c0,6.68-5.426,12.105-12.105,12.105H384.59v-25.724h31.528V194.694H384.59v-56.489h20.93 V36.321H187.625v43.361h-67.583v-49.42c0-6.682,5.423-12.105,12.105-12.105H461.01C467.695,18.158,473.109,23.581,473.109,30.263z M343.989,51.453h30.269v31.321c-3.18-1.918-6.868-3.092-10.853-3.092h-19.416V51.453z M394.177,232.021h-9.581v-12.105h9.581 V232.021z M384.59,262.284h9.581v12.105h-9.581V262.284z M384.59,303.14h9.581v12.104h-9.581V303.14z"/></svg>', () => {
+            this.disksMenuOpen = !this.disksMenuOpen;
+            diskButton[1].classList.toggle("ejs_svg_rotate", this.disksMenuOpen);
+            this.disksMenu.style.display = this.disksMenuOpen ? "" : "none";
+            diskButton[2].classList.toggle("ejs_disks_text", this.disksMenuOpen);
+        }, this.diskParent, true);
+        this.elements.menu.appendChild(this.diskParent);
+        this.closeDisksMenu = () => {
+            if (!this.disksMenu) return;
+            this.disksMenuOpen = false;
+            diskButton[1].classList.toggle("ejs_svg_rotate", this.disksMenuOpen);
+            diskButton[2].classList.toggle("ejs_disks_text", this.disksMenuOpen);
+            this.disksMenu.style.display = "none";
+        }
+        this.addEventListener(this.elements.parent, "mousedown touchstart", (e) => {
+            if (this.isChild(this.disksMenu, e.target)) return;
+            if (e.pointerType === "touch") return;
+            if (e.target === diskButton[0] || e.target === diskButton[2]) return;
+            this.closeDisksMenu();
+        })
+        this.addEventListener(this.canvas, "click", (e) => {
```

Comment:
Is this new?

Reply:
Yes... it's there to close the disks menu.

Reply:
The disks menu is a copy/paste of the settings menu - just with the word "Settings" replaced with "Disks" :D

Reply:
It appears to also attempt to lock the pointer. Did you remove the trigger from the other place?

Reply:
> The disks menu is a copy/paste of the settings menu - just with the word "Settings" replaced with "Disks"

Oh lol... didn't think I made it that modular lol nice

Reply:
> It appears to also attempt to lock the pointer. Did you remove the trigger from the other place?

It shouldn't... I reverted the change in this PR back to the original behaviour.
<img width="740" alt="Screenshot 2024-03-04 at 12 10 03 pm" src="https://github.com/EmulatorJS/EmulatorJS/assets/84688932/6b550b42-0adf-4842-9a21-e139c4f49195">
I'll double check what's been submitted to make sure though.

Reply:
> > The disks menu is a copy/paste of the settings menu - just with the word "Settings" replaced with "Disks"
> 
> Oh lol... didn't think I made it that modular lol nice

haha... there is probably an opportunity for refactoring down the track to make menu's from the toolbar button upwards, but for now, this'll do.

Reply:
I'm just confused why all the lines show as added, did your change get pushed to upstream? I never saw any pointer lock function calls removed

Reply:
Ah... that was my bad... I accidentally committed the pointer lock changes to wrong branch. Didn't notice until I was already deep in it, so just reverted it manually - that's why it shows as added.

Reply:
Ah ok, so that needs to be changed or is this ready to merge?

Reply:
Ready to merge. When I run this code locally, it doesn't capture my pointer, and behaves as the current production version does.

Reply:
It appears to be duplicated?

https://github.com/EmulatorJS/EmulatorJS/blob/e11b5fc7748463d1c71e0e1ee57fa104a41253de/data/emulator.js#L1911

https://github.com/EmulatorJS/EmulatorJS/blob/e11b5fc7748463d1c71e0e1ee57fa104a41253de/data/emulator.js#L1945

Reply:
You're right... I think I included it there because it needed to be there for each menu. I'll strip it out and test.

Reply:
Yup... works fine with just the one.

This is why you don't code at 2:30am folks :D

Reply:
That looks better...
<img width="913" alt="Screenshot 2024-03-04 at 12 30 22 pm" src="https://github.com/EmulatorJS/EmulatorJS/assets/84688932/41dbefd3-df63-4a03-a0ac-a6cca19f8210">

Reply:
I put a new line in between to make it clear it's got nothing to do with the settings menu.